tests/unit/test_s3.py

```diff
@@ -235,20 +169,50 @@ def test_append_last_modified_headers(self):
         s3_listener.append_last_modified_headers(response)
         self.assertNotEqual('No header', response.headers.get('Last-Modified', 'No header'))
 
-    def test_path_addressing_enabled_hosts(self):
-        headers = [
-            ({HEADER_LOCALSTACK_EDGE_URL: f'https://{HOSTNAME}:12345'}, True),
-            ({HEADER_LOCALSTACK_EDGE_URL: f'https://{HOSTNAME_EXTERNAL}:12345'}, True),
-            ({HEADER_LOCALSTACK_EDGE_URL: f'https://{LOCALHOST_IP}:12345'}, True),
-            ({'host': f'{HOSTNAME}:12345'}, True),
-            ({'host': f'{HOSTNAME_EXTERNAL}:12345'}, True),
-            ({'host': f'{LOCALHOST_IP}:12345'}, True),
-            ({'host': f'https://{HOSTNAME}:12345'}, False),
-            ({'host': f'https://{HOSTNAME_EXTERNAL}:12345'}, False),
-            ({'host': f'https://{LOCALHOST_IP}:12345'}, False),
+
+class S3UtilsTest (unittest.TestCase):
+
+    def test_s3_bucket_name(self):
+        # array description : 'bucket_name', 'expected_ouput'
+        bucket_names = [
+            ('docexamplebucket1', True),
+            ('log-delivery-march-2020', True),
+            ('my-hosted-content', True),
+            ('docexamplewebsite.com', True),
+            ('www.docexamplewebsite.com', True),
+            ('my.example.s3.bucket', True),
+            ('doc_example_bucket', False),
+            ('DocExampleBucket', False),
+            ('doc-example-bucket-', False)
+        ]
+
+        for bucket_name, expected_result in bucket_names:
+            assert expected_result == s3_utils.validate_bucket_name(bucket_name)
+
+    def test_bucket_name(self):
+        # array description : 'path', 'header', 'expected_ouput'
+        bucket_names = [
+            ('/bucket/keyname', {'host': f'https://{HOSTNAME}:4566'}, 'bucket'),
+            ('/bucket//keyname', {'host': f'https://{HOSTNAME}:4566'}, 'bucket'),
+            ('/keyname', {'host': f'bucket.{S3_VIRTUAL_HOSTNAME}:4566'}, 'bucket'),
+            ('//keyname', {'host': f'bucket.{S3_VIRTUAL_HOSTNAME}:4566'}, 'bucket'),
+            ('/', {'host': f'{S3_VIRTUAL_HOSTNAME}:4566'}, None)
         ]
-        for example_header, expected_result in headers:
-            assert expected_result == s3_listener.uses_path_addressing(example_header)
+
+        for path, headers, expected_result in bucket_names:
+            assert expected_result == s3_utils.extract_bucket_name(headers, path)
```

Comment:
nit: Same as above - let's better use `self.assertEqual(...)` (also in line 215)